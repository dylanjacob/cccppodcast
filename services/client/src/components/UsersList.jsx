import React from 'react';

const UsersList = (props) => {
  return (
    <div>
      <h2>All Users</h2>
      <table className="table">
        <thead>
          <tr>
            <th scope="col">#</th>
            <th scope="col">ID</th>
            <th scope="col">Username</th>
            <th scope="col">Email</th>
            <th scope="col">Active</th>
            <th scope="col">Admin</th>
          </tr>
        </thead>
        <tbody>
      {
        props.users.map((user, i) => {
          return (
                <tr id="user" key={ user.id }>
                  <th scope="row">{ i + 1 }</th>
                  <td>{ user.id }</td>
                  <td>{ user.username }</td>
                  <td>{ user.email }</td>
                  <td>{ user.active ? 'YES' : 'NO'  }</td>
                  <td>{ user.admin ? 'YES' : 'NO' }</td>
                </tr>
          )
        })
      }
        </tbody>
      </table>
    </div>
  )
};

export default UsersList;