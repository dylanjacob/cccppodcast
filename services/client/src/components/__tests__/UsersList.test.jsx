import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import UsersList from '../UsersList';

const users = [
  {
    'admin': true,
    'active': true,
    'email': 'dylan.jacob@bubblecore.net',
    'id': 1,
    'username': 'djacob'
  },
  {
    'admin': false,
    'active': true,
    'email': 'kwrussel@gmail.com',
    'id': 2,
    'username': 'kevin'
  }
];

test('UsersList renders properly', () => {
  const wrapper = shallow(<UsersList users={users}/>);
  expect(wrapper.find('h2').get(0).props.children).toBe('All Users');
  
  const table = wrapper.find('table');
  expect(table.length).toBe(1);

  expect(wrapper.find('thead').length).toBe(1);
  const th = wrapper.find('th[scope="col"]');
  expect(th.length).toBe(6);
  expect(th.get(0).props.children).toBe('#');
  expect(th.get(1).props.children).toBe('ID');
  expect(th.get(2).props.children).toBe('Username');
  expect(th.get(3).props.children).toBe('Email');
  expect(th.get(4).props.children).toBe('Active');
  expect(th.get(5).props.children).toBe('Admin');

  expect(wrapper.find('tbody').length).toBe(1);
  expect(wrapper.find('tbody > tr').length).toBe(2);
  const td = wrapper.find('tbody > tr > td');
  expect(td.length).toBe(10);
 
  expect(td.get(0).props.children).toBe(1);
  expect(td.get(1).props.children).toBe('djacob');
  expect(td.get(2).props.children).toBe('dylan.jacob@bubblecore.net');
  expect(td.get(3).props.children).toBe('YES');
  expect(td.get(4).props.children).toBe('YES');
  expect(td.get(5).props.children).toBe(2);
  expect(td.get(6).props.children).toBe('kevin');
  expect(td.get(7).props.children).toBe('kwrussel@gmail.com');
  expect(td.get(8).props.children).toBe('YES');
  expect(td.get(9).props.children).toBe('NO');
});

test('UsersList renders a snapshot properly', () => {
  const tree = renderer.create(<UsersList users={users}/>).toJSON();
  expect(tree).toMatchSnapshot();
});