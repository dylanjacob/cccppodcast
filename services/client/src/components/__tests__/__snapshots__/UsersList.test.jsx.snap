// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UsersList renders a snapshot properly 1`] = `
<div>
  <h2>
    All Users
  </h2>
  <table
    className="table"
  >
    <thead>
      <tr>
        <th
          scope="col"
        >
          #
        </th>
        <th
          scope="col"
        >
          ID
        </th>
        <th
          scope="col"
        >
          Username
        </th>
        <th
          scope="col"
        >
          Email
        </th>
        <th
          scope="col"
        >
          Active
        </th>
        <th
          scope="col"
        >
          Admin
        </th>
      </tr>
    </thead>
    <tbody>
      <tr
        id="user"
      >
        <th
          scope="row"
        >
          1
        </th>
        <td>
          1
        </td>
        <td>
          djacob
        </td>
        <td>
          dylan.jacob@bubblecore.net
        </td>
        <td>
          YES
        </td>
        <td>
          YES
        </td>
      </tr>
      <tr
        id="user"
      >
        <th
          scope="row"
        >
          2
        </th>
        <td>
          2
        </td>
        <td>
          kevin
        </td>
        <td>
          kwrussel@gmail.com
        </td>
        <td>
          YES
        </td>
        <td>
          NO
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
